packages/base-map/src/index.js

```diff
@@ -0,0 +1,259 @@
+                      detectRetina
+                    />
+                  )}
+                </LayersControl.BaseLayer>
+              );
+            })}
+
+          {/* user-controlled overlay layers (e.g., vehicle locations, stops) */}
+          {userControlledOverlays.map((child, i) => (
+            <LayersControl.Overlay
+              key={i}
+              name={child.props.name}
+              checked={child.props.visible}
+            >
+              {child}
+            </LayersControl.Overlay>
+          ))}
+        </LayersControl>
+
+        {/* Add the fixed, i.e. non-user-controllable, overlays (e.g., itinerary overlay) */}
+        {fixedOverlays}
+
+        {/* Add the location selection popup and content, if both are set. */}
+        {popupLocation && popupContent && (
+          <Popup position={popupLocation} onClose={onPopupClosed}>
+            {popupContent}
+          </Popup>
+        )}
+      </Map>
+    );
+  }
+}
+
+BaseMap.propTypes = {
+  children: PropTypes.oneOfType([
+    PropTypes.object,
+    PropTypes.arrayOf(PropTypes.object)
+  ]),
+  /**
+   * The configuration properties for the map.
+   */
+  mapConfig: PropTypes.shape({
+    baseLayers: PropTypes.arrayOf(
+      PropTypes.shape({
+        name: PropTypes.string.isRequired,
+        url: PropTypes.string.isRequired,
+        subdomains: PropTypes.string,
+        attribution: PropTypes.string,
+        maxZoom: PropTypes.number,
+        hasRetinaSupport: PropTypes.bool
+      })
+    ),
+    initLat: PropTypes.number,
+    initLon: PropTypes.number,
+    initZoom: PropTypes.number,
+    maxZoom: PropTypes.number
+  }),
+  /**
+   * Triggered when the user clicks on the map.
+   * See https://leafletjs.com/reference-1.6.0.html#map-click for details.
+   */
+  onClick: PropTypes.func,
+  /**
+   * Triggered when the user makes an overlay visible using the map's layers control.
+   * See https://leafletjs.com/reference-1.6.0.html#map-overlayadd for details.
+   */
+  onOverlayAdded: PropTypes.func,
+  /**
+   * Triggered when the user hides an overlay using the map's layers control.
+   * See https://leafletjs.com/reference-1.6.0.html#map-overlayremove for details.
+   */
+  onOverlayRemoved: PropTypes.func,
+  /**
+   * Triggered when the user closes the popup (if `popupLocation` and `popupContent` have been set).
+   */
+  onPopupClosed: PropTypes.func,
+  /**
+   * Triggered when the user pans the map or changes zoom level.
+   * See https://github.com/PaulLeCam/react-leaflet/blob/master/example/components/viewport.js for more details.
+   */
+  onViewportChanged: PropTypes.func,
+  /**
+   * The contents to render in the popup.
+   */
+  popupContent: PropTypes.node,
+  /**
+   * The coordinates of the popup object to display, as a [lat, lng] array.
+   */
+  popupLocation: PropTypes.arrayOf(PropTypes.number)
+};
+
+BaseMap.defaultProps = {
+  children: null,
+  mapConfig: {
+    initLat: 45.52,
+    initLon: -122.682,
```

Comment:
I wonder if we should not include default values for lat/lng/zoom and instead make these required props. @evansiroky?

Reply:
initZoom should remain optional.

Reply:
Agree w/ Landon about lat & lon being required.  Could be strange for an Atlanta map to zoom into Portland by default.  Zoom should have a default value.